mesonbuild/scripts/dist.py

```diff
@@ -0,0 +1,155 @@
+
+def create_dist(dist_name, src_root, bld_root, dist_sub):
+    distdir = os.path.join(dist_sub, dist_name)
+    gitdir = os.path.join(src_root, '.git')
+    if os.path.exists(distdir):
+        shutil.rmtree(distdir)
+    os.makedirs(distdir)
+    dest_gitdir = os.path.join(distdir, '.git')
+    try:
+        os.symlink(gitdir, dest_gitdir)
+    except Exception:
+        # Could not symlink, must copy.
+        shutil.copytree(gitdir, dest_gitdir)
+    subprocess.check_call(['git', 'reset', '--hard'], cwd=distdir)
+    process_submodules(distdir)
+    del_gitfiles(distdir)
+    xzname = distdir + '.tar.xz'
+    zipname = distdir + '.zip'
+    # Should use shutil but it got xz support only in 3.5.
+    with tarfile.open(xzname, 'w:xz') as tf:
+        tf.add(distdir, os.path.split(distdir)[1])
+    # Create only .tar.xz for now.
+    #create_zip(zipname, distdir)
+    shutil.rmtree(distdir)
+    return (xzname, )
+
+def check_dist(packagename, meson_command):
+    print('Testing distribution package %s.' % packagename)
+    unpackdir = tempfile.mkdtemp()
+    builddir = tempfile.mkdtemp()
+    installdir = tempfile.mkdtemp()
+    ninja_bin = detect_ninja()
+    try:
+        tf = tarfile.open(packagename)
+        tf.extractall(unpackdir)
+        srcdir = glob(os.path.join(unpackdir, '*'))[0]
+        if subprocess.call(meson_command + [srcdir, builddir]) != 0:
+            print('Running Meson on distribution package failed')
+            return 1
+        if subprocess.call([ninja_bin], cwd=builddir) != 0:
+            print('Compiling the distribution package failed.')
+            return 1
+        if subprocess.call([ninja_bin, 'test'], cwd=builddir) != 0:
+            print('Running unit tests on the distribution package failed.')
+            return 1
+        myenv = os.environ.copy()
+        myenv['DESTDIR'] = installdir
+        if subprocess.call([ninja_bin, 'install'], cwd=builddir, env=myenv) != 0:
+            print('Installing the distribution package failed.')
+            return 1
+    finally:
+        shutil.rmtree(srcdir)
+        shutil.rmtree(builddir)
+        shutil.rmtree(installdir)
+    print('Distribution package %s tested.' % packagename)
+    return 0
+
+def run(args):
+    src_root = args[0]
+    bld_root = args[1]
+    meson_command = args[2:]
+    priv_dir = os.path.join(bld_root, 'meson-private')
+    dist_sub = os.path.join(bld_root, 'meson-dist')
+
+    buildfile = os.path.join(priv_dir, 'build.dat')
+
+    build = pickle.load(open(buildfile, 'rb'))
+
+    dist_name = build.project_name + '-' + build.project_version
+
+    if not os.path.isdir(os.path.join(src_root, '.git')):
+        print('Dist currently only works with Git repos.')
+        return 1
+    names = create_dist(dist_name, src_root, bld_root, dist_sub)
+    if names is None:
+        return 1
+    error_count = 0
+    for name in names:
+        #rc = check_dist(name, meson_command) # Check only one.
```

Comment:
__[flake8]__
* [E265] block comment should start with '# '

<!-- Comment by [SideCI](https://sideci.com) -->